jenkins/Jenkinsfile.nightly

```
@@ -0,0 +1,244 @@
+								dir('./services/blockchain-indexer') { sh 'npm ci' }
+								dir('./services/blockchain-coordinator') { sh 'npm ci' }
+								dir('./services/core') { sh 'npm ci' }
+								dir('./services/fee-estimator') { sh 'npm ci' }
+								dir('./services/market') { sh 'npm ci' }
+								dir('./services/newsfeed') { sh 'npm ci' }
+								dir('./services/export') { sh 'npm ci' }
+								dir('./services/gateway') { sh 'npm ci' }
+								dir('./services/template') { sh 'npm ci' }
+								dir('./services/transaction-statistics') { sh 'npm ci' }
+								dir('./tests') { sh 'npm ci' }
+							}
+						}
+					}
+				}
+			}
+		}
+		stage('Run required services') {
+			when {
+				anyOf {
+					branch 'development'
+					branch 'feature/blockchain-agnostic'
+				}
+			}
+			parallel {
+				stage('Lisk Core') {
+					steps {
+						script { echoBanner(STAGE_NAME) }
+						dir('lisk-core') {
+							nvm(readFile('.nvmrc').trim()) {
+								sh '''
+									./bin/run start --network devnet --api-ipc &
+									echo $! > lisk-core.pid
+								'''
+							}
+						}
+					}
+				}
+				stage('Lisk Service') {
+					steps {
+						dir('lisk-service') {
+							script {
+								runServiceIfMissing('MySQL', './jenkins/mysql', MYSQL_PORT)
+								runServiceIfMissing('Redis', './jenkins/redis', REDIS_PORT)
+							}
+						}
+					}	
+				}
+			}
+		}
+		stage('Run microservices') {
+			when {
+				anyOf {
+					branch 'development'
+					branch 'feature/blockchain-agnostic'
+				}
+			}
+			steps {
+				script { echoBanner(STAGE_NAME) }
+				dir('lisk-service') {
+					nvm(readFile('.nvmrc').trim()) {
+						sh 'pm2 start --silent ecosystem.jenkins.config.js' 
+					}
+					waitForHttp('http://localhost:9901/api/v3/blocks')
+				}
+			}
+		}
+		stage('Run Tests') {
+			when {
+				anyOf {
+					branch 'development'
+					branch 'feature/blockchain-agnostic'
+				}
+			}
+			parallel {
+				stage ('Check linting') {
+					steps {
+						script { echoBanner(STAGE_NAME) }
+						dir('lisk-service') {
+							nvm(readFile('.nvmrc').trim()) {
+								sh 'npm run eslint'
+							}
+						}
+					}
+				}
+				stage('Run functional tests') {
+					steps {
+						script { echoBanner(STAGE_NAME) }
+						dir('lisk-service') {
+							nvm(readFile('.nvmrc').trim()) {
+								dir('./framework') { sh "npm run test:functional" }
+								dir('./services/market') { sh "npm run test:functional" }
+								dir('./tests') { sh "npm run test:functional" }
+							}
+						}
+					}
+				}
+				stage('Run integration tests') {
+					steps {
+						script { echoBanner(STAGE_NAME) }
+						dir('lisk-service') {
+							nvm(readFile('.nvmrc').trim()) {
+								dir('./tests') { sh 'npm run test:integration:APIv3:SDKv6' }
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+	post {
+		failure {
+			script { echoBanner('Failed to run the pipeline') }
+			nvm(readFile('.nvmrc').trim()) {
+				sh 'pm2 logs'	
+			}
```

```suggestion
			script { 
			    echoBanner('Failed to run the pipeline')
			    dir('lisk-service') {
			        nvm(readFile('.nvmrc').trim()) {
				        sh 'pm2 logs'	
			        }
			    }
			}
```

Not sure why the nightly job fails, but this part is failing for not running in the lisk-service directory (my whitespace might be off, I'd look add this in manually and double check it instead of accepting my suggestion as is :sweat_smile:)